pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -366,6 +373,7 @@ func prepareSeedConfig(ctx context.Context, gardenClient client.Client, seedClie
 			BlockCIDRs:    shootedSeedConfig.BlockCIDRs,
 			ShootDefaults: shootedSeedConfig.ShootDefaults,
 		},
+		IngressController: shootedSeedConfig.IngressSettings,
```

Comment:
This can be computed by us: If the shoot's nginx-ingress addon is disabled, then we enable this here. If the shoot's nginx-addon is enabled, then we disable this here.
Then you can overwrite these computations with what was provided the `shootedSeedConfig`.